edb/server/server.py

```diff
@@ -351,11 +356,22 @@ def get_compilation_system_config(self):
         return self._dbindex.get_compilation_system_config()
 
     async def acquire_pgcon(self, dbname):
-        return await self._pg_pool.acquire(dbname)
+        if self._pg_unavailable_msg is not None:
+            raise errors.PostgresUnavailableError(
+                'Postgres is not available: ' + self._pg_unavailable_msg
+            )
+
+        while True:
```

Comment:
Should we have a limit here?

Reply:
Sure! `_pg_pool.max_capacity + 1` sounds like a reasonable limit